discord/soundboard.py

```diff
@@ -0,0 +1,271 @@
+
+    .. versionadded:: 2.7
+
+    Attributes
+    ----------
+    id: :class:`int`
+        The sound's ID.
+    volume: :class:`float`
+        The sound's volume.
+    emoji: :class:`PartialEmoji` | :class:`None`
+        The sound's emoji. Could be ``None`` if the sound has no emoji.
+    """
+
+    __slots__ = ("id", "volume", "emoji", "_http", "_state")
+
+    def __init__(
+        self,
+        data: SoundboardSoundPayload | VoiceChannelEffectSendEventPayload,
+        state: ConnectionState,
+        http: HTTPClient,
+    ):
+        self._http = http
+        self._state = state
+        self._from_data(data)
+
+    def _from_data(
+        self, data: SoundboardSoundPayload | VoiceChannelEffectSendEventPayload
+    ) -> None:
+        self.id = int(data.get("sound_id", 0))
+        self.volume = (
+            float(data.get("volume", 0) or data.get("sound_volume", 0)) or None
+        )
+        self.emoji = None
+        if raw_emoji := data.get(
+            "emoji"
+        ):  # From gateway event (VoiceChannelEffectSendEventPayload)
+            self.emoji = PartialEmoji.from_dict(raw_emoji)
+        elif data.get("emoji_name") or data.get(
+            "emoji_id"
+        ):  # From HTTP response (SoundboardSoundPayload)
+            self.emoji = PartialEmoji(
+                name=data.get("emoji_name"),
+                id=int(data.get("emoji_id", 0) or 0) or None,
+            )
+
+    @override
+    def __eq__(
+        self, other: PartialSoundboardSound
+    ) -> bool:  # pyright: ignore[reportIncompatibleMethodOverride]
+        if isinstance(other, self, __class__):
+            return self.id == other.id
+        return NotImplemented
+
+    @override
+    def __ne__(
+        self, other: PartialSoundboardSound
+    ) -> bool:  # pyright: ignore[reportIncompatibleMethodOverride]
+        return not self.__eq__(other)
+
+    @property
+    def file(self) -> Asset:
+        """:class:`Asset`: Returns the sound's file."""
+        return Asset._from_soundboard_sound(self._state, sound_id=self.id)
+
+    def __repr__(self) -> str:
+        return f"<PartialSoundboardSound id={self.id} volume={self.volume} emoji={self.emoji!r}>"
+
+
+class SoundboardSound(PartialSoundboardSound):
+    """Represents a soundboard sound.
+
+    .. versionadded:: 2.7
+
+    Attributes
+    ----------
+    id: :class:`int`
+        The sound's ID.
+    volume: :class:`float`
+        The sound's volume.
+    emoji: :class:`PartialEmoji` | :class:`None`
+        The sound's emoji. Could be ``None`` if the sound has no emoji.
+    name: :class:`str`
+        The sound's name.
+    available: :class:`bool`
+        Whether the sound is available. Could be ``False`` if the sound is not available.
+        This happens for example when the guild lost the boost level required to use the sound.
+    guild_id: :class:`int` | :class:`None`
+        The ID of the guild the sound belongs to. Could be :class:`None` if the sound is a default sound.
```

Comment:
Should not end with a preposition.
```suggestion
        The ID of the guild to which the sound belongs. Could be :class:`None` if the sound is a default sound.
```